File: LeetCode/Study Plan/Practice Problems/Easy/Make The String Great/Go/make_the_string_great.go
package main

import (
	"fmt"
	"strings"
)

func make_the_string_great(s string) string {
	stack := []byte{}
	n := len(s)
	for i := 0; i < n; i++ {
		if len(stack) == 0 {
			stack = append(stack, s[i])
		} else {
			top := stack[len(stack)-1]
			curr := s[i]
			if strings.ToLower(string(top)) == strings.ToLower(string(curr)) {
				if int(curr)-int(top) != 0 {
					stack = stack[:len(stack)-1]
				} else {
					stack = append(stack, s[i])
				}
			} else {
				stack = append(stack, s[i])
			}
		}
	}
	return string(stack)
}

func main() {
	fmt.Println(make_the_string_great("leEeetcode"))
}

File: LeetCode/Easy/Make The String Great/Cpp/make_the_string_great.cc
#include <iostream>
#include <stack>

typedef std::stack<char> sc;

std::string make_the_string_great(std::string st) {
    sc s;
    for (int i=0; i<st.length(); i++) {
        if (s.empty()) s.push(st[i]);
        else {
            char t = s.top();
            char curr = st[i];
            if (tolower(t) == tolower(curr)) {
                if ((t-curr)!=0) s.pop();
                else s.push(st[i]);
            } else {
                s.push(st[i]);
            }
        }
    }
    std::string res = "";
    while(!s.empty()) {
        res = s.top() + res;
        s.pop();
    }
    return res;
}

int main() {
    std::cout<<make_the_string_great("leEeetcode")<<"\n";
}
